### PR TITLE
Front-end pull stream

### PR DESCRIPTION
**Pull request by aquaopaque**

### CODE REVIEW

**Comment by gregwym** [comment]
Could you please add some screenshot? And also explain how the registration, login and checkout process works? In either English or Chinese.

**Comment by gregwym** [comment]
Ping me when you update this. Or you wanna talk over QQ.

**Comment by aquaopaque** [comment]
Login/Logout prcoess on the checkout page.

Login process is tied up with delivery proccess. On delivery, there are options for login or register. 
1) Login button will initiate the login process
2) Register button provides user to enter user name and password, beside the delivery info. When user proceeds checkout, the information for delivery is used for user sign up. 
Once the user logged in, the delivery page shows two additional optoins
1) My Profile takes user to an individual profile view page. Where user can choose update profile or password or logout.
2) Logout button clears the user session and take user back to cart viewing page.

File: account/migrations/0002_auto__del_field_userprofile_phone_num__add_field_userprofile_phone__ad.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+        # Deleting field 'UserProfile.phone_num'
+        db.delete_column(u'account_userprofile', 'phone_num')
+
+        # Adding field 'UserProfile.phone'
+        db.add_column(u'account_userprofile', 'phone',
+                      self.gf('django.db.models.fields.CharField')(default='', max_length=20),
+                      keep_default=False)
```
Comment (gregwym) [inline account/migrations/0002_auto__del_field_userprofile_phone_num__add_field_userprofile_phone__ad.py]: Checkout this, http://stackoverflow.com/questions/3235995/django-how-to-rename-a-model-field-using-south

**Comment by gregwym** [comment]
Clicking `enter` in the login form close the form instead of login. I think it is the same case for the register form.

**Comment by gregwym** [comment]
qq20140210-1 2x

The error message's location is kinda wired, but let leave it like this for now. We can fix this when we have a new design for the entire site. I'm just documenting it here.

**Comment by gregwym** [comment]
Can we login the user right after registration? So he/she doesn't have to login again when he/she come back next time.

**Comment by gregwym** [comment]
I didn't test this, but think you might know. Does Django Auth enforce unique email registration?

**Comment by gregwym** [comment]
On all the pages, lets try to avoid js `history.back()`. 
- Profile page: change it to `Go back to cart`. (Even better if it can remember where it came from. Passing a `back` property as GET query, and use cart url as default if the property does not exist. )
- Profile update and password change: change it to `Go back to profile`

**Comment by gregwym** [comment]
There is a `user` field in `Invoice` model. Please fill-in that field in `create_invoice`.

**Comment by gregwym** [comment]
I think we need a `Forgot password` besides the `Login` button. 
Also a line-note in the register form that tell customer, "You account will be created after checkout".

File: ntemplates/order/cart.html [hunk 2]
```diff
@@ -32,7 +32,8 @@
         <div class="item-info-left">
           <div class="item-in-cart-name">{{ item.obj.name }}</div>
           <div class="item-in-cart-store">From: {{ item.obj.category.store.name }}</div>
-          <div><label><input type="checkbox" name="allow_sub_detail[{{ item.obj.id }}]" checked="">Allow substitution</label></div>
+          <div title="If checked, the delivery person will pick up alternative item for this item. Or you'll get refund on delivery">
+            <label><input type="checkbox" name="allow_sub_detail[{{ item.obj.id }}]" checked="">Allow substitution</label></div>
```
Comment (gregwym) [inline ntemplates/order/cart.html]: Please move that `</div>` into a new line.

File: ntemplates/order/checkout.html [hunk 3]
```diff
@@ -154,6 +221,8 @@
   </table>
 </script>
 <script type="text/javascript">
+var account_check = {};
```
Comment (gregwym) [inline ntemplates/order/checkout.html]: Move this variable into the `.ready(function() {...})` scope. The rule is, avoid global variable.

File: ntemplates/order/checkout.html [hunk 4]
```diff
@@ -161,6 +230,168 @@
...
+          errors.html("<p>Login failed. Please try again later or report this account to us. Thank you</p>");
+          errors.show();
+          // it's highly possible the iframe is not in login page any more.
+          ifm.prop('src', ifm.prop('src'));
+        }
+      }
+    });
+  });
+
+  var isRegisterOk = function(cbfunc) {
```
Comment (gregwym) [inline ntemplates/order/checkout.html:276]: Interesting name for callback function, lol. Its the first time I see this.

File: ntemplates/order/checkout.html [hunk 5]
```diff
@@ -161,6 +230,168 @@
...
+      if(other.hasClass('active')) {
+        other.button('toggle');
+        otherPnl.hide();
+      }
+      thisPnl.show();
+      pnlContainer.slideDown();
+    }
+  });
+
+  var detect_user = function() {
```
Comment (gregwym) [inline ntemplates/order/checkout.html]: Please follow the same naming style, this should be `detectUser`.

File: ntemplates/order/checkout.html [hunk 6]
```diff
@@ -154,12 +221,188 @@
...
+    });
+  };
+
+  $("#login-id_password").keypress(function(evt){
+    if(evt.which == 13) {
+      $("#login-submit").click();
+      evt.preventDefault();
+    }
+    //evt.stopPropagation();
+  });
```
Comment (gregwym) [inline ntemplates/order/checkout.html:345]: Isn't the `login` section actually a form? I thought press enter within a form will trigger submit directly.

**Pull request merged**